In [1]:
import numpy as np

In [2]:
a = np.random.random(int(1e8))

In [3]:
import time
start = time.time()
sum(a) / len(a)
print(time.time() - start, 'seconds')

8.496863842010498 seconds


In [4]:
start = time.time()
np.mean(a)
print(time.time() - start, 'seconds')

0.0486147403717041 seconds


Using Python takes nearly 8.5 seconds, while Numpy takes only 0.05 seconds.